src/ooni/utils.hpp

```diff
@@ -5,34 +5,86 @@
 #ifndef SRC_OONI_UTILS_HPP
 #define SRC_OONI_UTILS_HPP
 
+#include "src/ext/json/src/json.hpp"
+#include <GeoIP.h>
 #include <measurement_kit/common.hpp>
 #include <measurement_kit/http.hpp>
 #include <regex>
 #include <string>
 
+using Json = nlohmann::json;
+
 namespace mk {
 namespace ooni {
 
 template <decltype(mk::http::get) httpget = mk::http::get>
 void ip_lookup(Callback<std::string> callback) {
     httpget("http://geoip.ubuntu.com/lookup",
-              [=](Error err, http::Response response) {
-                  if (err) {
-                      callback(err, "");
-                      return;
-                  }
-                  if (response.status_code != 200) {
-                      callback(GenericError(), "");
-                      return;
-                  }
-                  std::smatch m;
-                  std::regex regex("<Ip>(.*)</Ip>");
-                  if (std::regex_search(response.body, m, regex) == false) {
-                      callback(GenericError(), "");
-                      return;
-                  }
-                  callback(NoError(), m[1]);
-              }, {}, "", {}, Logger::global(), Poller::global());
+            [=](Error err, http::Response response) {
+                if (err) {
+                    callback(err, "");
+                    return;
+                }
+                if (response.status_code != 200) {
+                    callback(GenericError(), "");
+                    return;
+                }
+                std::smatch m;
+                std::regex regex("<Ip>(.*)</Ip>");
+                if (std::regex_search(response.body, m, regex) == false) {
+                    callback(GenericError(), "");
+                    return;
+                }
+                callback(NoError(), m[1]);
+            },
+            {}, "", {}, Logger::global(), Poller::global());
+}
+
+// this is needed to return a const char instead of a char for asn resolver
+inline GEOIP_API const char *
```

Comment:
Here GEOIP_API is not needed. (This GEOIP_API is a compiler definition used when building GeoIP that basically tells the compiler that the following function shall be exported. It is typically a no op on Unix systems and has meaning on Windows. As such, it may even break our build if we attempt building on Win).